File: src/main/java/plus/maa/backend/controller/request/user/RegisterDTO.java
package plus.maa.backend.controller.request.user;

import jakarta.validation.constraints.Email;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;
import org.hibernate.validator.constraints.Length;

/**
 * @author AnselYuki
 */
@Data
@Accessors(chain = true)
@NoArgsConstructor
@AllArgsConstructor
public class RegisterDTO {
    @Email(message = "邮箱格式错误")
    private String email;
    @Length(min = 4, max = 24, message = "用户名长度应在2-24位之间")
    private String userName;
    @Length(min = 8, max = 32, message = "密码长度必须在8-32位之间")
    private String password;
    private String registrationToken;
}

File: src/main/kotlin/plus/maa/backend/controller/request/user/RegisterDTO.kt
package plus.maa.backend.controller.request.user

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import org.hibernate.validator.constraints.Length

/**
 * @author AnselYuki
 */
data class RegisterDTO(
    @field:NotBlank(message = "邮箱格式错误")
    @field:Email(message = "邮箱格式错误")
    val email: String,
    @field:NotBlank(message = "用户名长度应在4-24位之间")
    @field:Length(min = 4, max = 24, message = "用户名长度应在4-24位之间")
    val userName: String,
    @field:NotBlank(message = "密码长度必须在8-32位之间")
    @field:Length(min = 8, max = 32, message = "密码长度必须在8-32位之间")
    val password: String,
    @field:NotBlank(message = "请输入验证码")
    val registrationToken: String,
)
